### PR TITLE
FED-3255 Allow analyzer 6.x

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
Depends on https://github.com/Workiva/over_react/pull/958

## Motivation
over_react currently allows analyzer 5.x, but not analyzer 6.x, which prevents newer versions of other packages from being pulled in.

We were previously blocked on supporting analyzer 6.x since it can only be used in Dart 3, but now we support Dart 3!

## Changes
- Raise analyzer upper bound to allow 6.x
- Add 6.x to CI matrix that validates behavior under different analyzer majors
- Fix usages of deprecated `NamedType.name`, which was removed in analyzer 6.0.0
    - This getter included the name and any import prefix, and was deprecated in favor of the following getters:
        - `name2` (the name without the prefix)
        - `importPrefix` (just the prefix)

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - CI passes
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/vm_tests/builder/parsing/ast_util_test.dart [hunk 1]
```diff
@@ -152,17 +152,28 @@ main() {
...
+          SomeTypeName foo;
+        ''')).nameWithPrefix, 'SomeTypeName');
+
+        expect(TypeNameHelper(parseAndGetSingleWithType('''
+          foo.SomeTypeName foo;
+        ''')).nameWithPrefix, 'foo.SomeTypeName');
+      });
     });
 
     group('NameHelper', () {
```
Comment (greglittlefield-wf) [inline test/vm_tests/builder/parsing/ast_util_test.dart:167]: This extension is exclusive to `Identifier`, and in analyzer 6 type names are no longer parsed as `Identifier`, so these cases needed to be updated

**Review by robbecker-wf — DISMISSED** [review]
🎊

**Comment by robbecker-wf** [comment]
QA+1 CI passes on all combinations. Validated that analyzer 5 and 6 are actually used and pass.

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Pull request merged**